packages/explat-client/src/internal/experiment-assignments.ts

```diff
@@ -42,15 +27,18 @@ const fallbackExperimentAssignmentTtl = 60;
 
 /**
  * A fallback ExperimentAssignment we return when we can't retrieve one.
+ * As it is used in fallback situations, this function must never throw.
  *
  * @param experimentName The name of the experiment
  */
 export const createFallbackExperimentAssignment = (
 	experimentName: string
-): ExperimentAssignment => ( {
-	experimentName,
-	variationName: null,
-	retrievedTimestamp: Timing.monotonicNow(),
-	ttl: fallbackExperimentAssignmentTtl,
-	isFallbackExperimentAssignment: true,
-} );
+): ExperimentAssignment => {
+	return {
+		experimentName: isName( experimentName ) ? experimentName : 'fallback_experiment_assignment',
+		variationName: null,
+		retrievedTimestamp: Timing.monotonicNow(),
+		ttl: fallbackExperimentAssignmentTtl,
+		isFallbackExperimentAssignment: true,
+	};
+};
```

Comment:
```suggestion
): ExperimentAssignment => ({
		experimentName: isName( experimentName ) ? experimentName : 'fallback_experiment_assignment',
		variationName: null,
		retrievedTimestamp: Timing.monotonicNow(),
		ttl: fallbackExperimentAssignmentTtl,
		isFallbackExperimentAssignment: true,
	});
};
```

Reply:
It would also benefit the reader to put these in some kind of order (perhaps by length?). I found playing "where's Waldo" with the `:` fairly annoying. Not the end of the world though.

Reply:
> It would also benefit the reader to put these in some kind of order (perhaps by length?). I found playing "where's Waldo" with the : fairly annoying. Not the end of the world though.

Ah I do use this order on purpose and everywhere (I hope) - ordered by "importance". Experiment Name, then Variation Name, then retrieved timestamp, then ttl, then (optionally) fallback status. This order will be kept when printing/debugging/developing (in typescript aware editors).